lib/rpmsg/rpmsg.c

```diff
@@ -110,5 +110,6 @@ void rpmsg_destroy_ept(struct rpmsg_endpoint *ept)
 		return;
 
 	rpmsg_unregister_endpoint(ept);
-	metal_free_memory(ept);
+	if (ept->destroy_cb)
+		ept->destroy_cb(ept);
```

Comment:
As rpmsg_create_ept dynamically allocated the ept,  the memory free seems missing for the destroy...
The result of the patch is that application has to free an ept that it does not create.
My opinion is that this is not coherent. Either application should  allocate and free ept structure or rpmsg should do it.

Reply:
Good point, I don't think I realized that rpmsg_create_ept allocated and ept.  Since that's the case will look to clean this up so we do the freeing as well.

Reply:
it depends how to use the  destroy_endpoint, if we allow endpoint to be statically created, which i think yes, destroy endpoint should not do free_endpoint.
I think we can document the pair of the functions.